pkg/importer/dataStream.go

```diff
@@ -266,6 +273,110 @@ func (d *DataStream) http() (io.ReadCloser, error) {
+	io.Reader
+}
+
+func (registryDataCloser) Close() error {
+	_regData.file.Close()
+	os.RemoveAll(_regData.dataDir)
+	return nil
+}
+
+// RegistryDataCloser returns a ReadCloser with a Close method wrapping delettion
+// of temporary data of registry flow
+func RegistryDataCloser(r io.Reader) io.ReadCloser {
+	return registryDataCloser{r}
+}
+
+const (
+	//ContainerDiskImageDir - Expected disk image location in contianer image as described in
+	//https://github.com/kubevirt/kubevirt/blob/master/docs/container-register-disks.md
+	ContainerDiskImageDir = "disk"
+	//TempContainerDiskDir - Temporary location for pulled containerImage
+	TempContainerDiskDir = "tmp"
+)
+
+//This import source downloads specified container image from registry location
+//Then it extracts the image to a temporary location and expects an image file to be located under /disk directory
+//If such exists it creates a Reader on it and returns it for further processing
+func (d *DataStream) registry() (io.ReadCloser, error) {
+	//generate random name for temporary directory
+	_regData.dataDir = filepath.Join(d.DataDir, TempContainerDiskDir)
+
+	imageDir := filepath.Join(_regData.dataDir, ContainerDiskImageDir)
+
+	//create temporary directory if note exists to which all the data will be extracted
+	if _, err := os.Stat(_regData.dataDir); os.IsNotExist(err) {
+		err := os.Mkdir(_regData.dataDir, os.ModeDir|os.ModePerm)
+		if err != nil {
+			return nil, errors.Wrapf(err, fmt.Sprintf("Failed to create tempdirectory %s", _regData.dataDir))
+		}
+	}
+
+	//copy image from registry to the temporary location
+	glog.V(1).Infof("using skopeo to copy from registry")
+	err := image.CopyDirFromRegistryImage(d.Endpoint, _regData.dataDir, ContainerDiskImageDir, d.AccessKey, d.SecKey)
+	if err != nil {
+		os.RemoveAll(_regData.dataDir)
+		return nil, errors.Wrapf(err, fmt.Sprintf("Failed ro read from registry"))
+	}
+
+	//Search for file in /disk directory - if not found - failure
+	imageFile, err := getImageFileName(imageDir)
+	if err != nil {
+		os.RemoveAll(_regData.dataDir)
+		return nil, errors.Wrapf(err, fmt.Sprintf("Cannot locate image file"))
+	}
+
+	// 3. If found - Create a reader that will read this file and attach it to the dataStream
+	_regData.file, err = os.Open(filepath.Join(imageDir, imageFile))
+	if err != nil {
+		os.RemoveAll(_regData.dataDir)
+		return nil, errors.Wrapf(err, fmt.Sprintf("Fail to create data stream from image file"))
+	}
+
+	glog.V(1).Infof("VM disk image filename is %s", imageFile)
+
+	return RegistryDataCloser(bufio.NewReader(_regData.file)), nil
+}
+
+func getImageFileName(dir string) (string, error) {
+
+	if _, err := os.Stat(dir); os.IsNotExist(err) {
+		return "", errors.Wrapf(err, fmt.Sprintf("image directory does not exist "))
+	}
+
+	entries, err := ioutil.ReadDir(dir)
+	if len(entries) == 0 || err != nil {
+		return "", errors.Wrapf(err, fmt.Sprintf("image file does not exist in image directory"))
+	}
+
+	fileinfo := entries[len(entries)-1]
```

Comment:
Should we maybe add a little more intelligence here if there are multiple files?  Like, if there happens to be multiple files, but one is named `disk.img` use that?  Or maybe inspect content types?  Or maybe it's an error?

Reply:
this logic is the same as in containerDisk kubevirt/cmd/container-disk-v1alpha/entry-point.sh. Our intention is to have the same logic. we shall not assume that the disk image file has specific name or format